test/Mono.Linker.Tests.Cases/Interop/PInvoke/Warnings/ComPInvokeWarning.cs

```diff
@@ -15,39 +15,175 @@ class ComPInvokeWarning
+			Call_CanSuppressWithRequiresUnreferencedCode ();
 		}
 
 		[ExpectedWarning ("IL2050")]
+		static void Call_SomeMethodTakingInterface ()
+		{
+			SomeMethodTakingInterface (null);
+		}
 		[DllImport ("Foo")]
 		static extern void SomeMethodTakingInterface (IFoo foo);
 
 		[ExpectedWarning ("IL2050")]
+		static void Call_SomeMethodTakingObject ()
+		{
+			SomeMethodTakingObject (null);
+		}
 		[DllImport ("Foo")]
 		static extern void SomeMethodTakingObject ([MarshalAs (UnmanagedType.IUnknown)] object obj);
 
 		[ExpectedWarning ("IL2050")]
+		static void Call_SomeMethodTakingArray ()
+		{
+			SomeMethodTakingArray (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingArray (Array array);
+
+		static void Call_SomeMethodTakingStringBuilder ()
+		{
+			SomeMethodTakingStringBuilder (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingStringBuilder (StringBuilder str);
+
+		static void Call_SomeMethodTakingCriticalHandle ()
+		{
+			SomeMethodTakingCriticalHandle (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingCriticalHandle (MyCriticalHandle handle);
+
+
+		static void Call_SomeMethodTakingSafeHandle ()
+		{
+			SomeMethodTakingSafeHandle (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingSafeHandle (TestSafeHandle handle);
+
+		static void Call_SomeMethodTakingExplicitLayoutClass ()
+		{
+			SomeMethodTakingExplicitLayout (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingExplicitLayout (ExplicitLayout _class);
+
+		static void Call_SomeMethodTakingSequentialLayoutClass ()
+		{
+			SomeMethodTakingSequentialLayout (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingSequentialLayout (SequentialLayout _class);
+
+		[ExpectedWarning ("IL2050")]
+		static void Call_SomeMethodTakingAutoLayoutClass ()
+		{
+			SomeMethodTakingAutoLayout (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingAutoLayout (AutoLayout _class);
+
+
+		static void Call_SomeMethodTakingString ()
+		{
+			SomeMethodTakingString (null);
+		}
+		[DllImport ("Foo")]
+		static extern void SomeMethodTakingString (String str);
+
+		[ExpectedWarning ("IL2050")]
+		static void Call_GetInterface ()
+		{
+			GetInterface ();
+		}
 		[DllImport ("Foo")]
 		static extern IFoo GetInterface ();
 
 		[UnconditionalSuppressMessage ("trim", "IL2050")]
+		static void Call_CanSuppressWarningOnParameter ()
+		{
+			CanSuppressWarningOnParameter (null);
+		}
 		[DllImport ("Foo")]
 		static extern void CanSuppressWarningOnParameter ([MarshalAs (UnmanagedType.IUnknown)] object obj);
 
 		[UnconditionalSuppressMessage ("trim", "IL2050")]
+		static void Call_CanSuppressWarningOnReturnType ()
+		{
+			CanSuppressWarningOnReturnType ();
+		}
 		[DllImport ("Foo")]
 		static extern IFoo CanSuppressWarningOnReturnType ();
 
-		[ExpectedWarning ("IL2050")] // Issue https://github.com/mono/linker/issues/1989
 		[RequiresUnreferencedCode ("test")]
```

Comment:
Can you please also add a test that RUC on the Pinvoke itself works as expected?
I know it's not exactly what this fix is about, but it's very related.